### DISCUSSION TITLE
reverse proxy mode disable http3

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#3461] [Ideas] [answered]
No reverse proxy I know can connect to a http3 backend (nearly all can't even connect to a http2 backend), so it would be good if http3 would be disabled in reverse proxy mode (and no udp would be exposed)

**Comment by FlattusBlastus** [comment]
Caddy can do it but its still not all that mature in browsers. Its slow and overly complex.
https://caddy.community/t/caddy-not-using-http3/19555/2

**Reply by Zoey2936** [reply]
I think you missunderstood this thing completly http3 is great, but in for example
browser with http3 support <=> AIO - http3 in AIO makes sense
browser with http3 support <=> reverse proxy which can only connect to http1.1 backend, but can serve http3 <=> AIO - http3 in AIO makes no sense, since the reverse proxy (apache, caddy, nginx, etc.) can only connect to backend server using http1.1, even if caddy and nginx can serve http3 to the clients, that is why it makes no sense to block an udp port which will never be used

**Comment by FlattusBlastus** [comment]
NGINX has it as well:
```
location / { 
    # advertise that QUIC is available on the configured port 
    add_header Alt-Svc 'h3=":$server_port"; ma=86400'; 
       
    # signal whether we are using QUIC+HTTP/3
    add_header X-protocol $server_protocol always;        
    
    #proxy_pass <upstream_group>; 
    #root /<root_directory>; 
}
```

**Comment by szaimen** [comment]
Hi @Zoey2936, are you fine if we just do not expose the udp port in reverse proxy mode? Disabling http3 in the caddyfile would add further complexity which I would like to avoid if possible...

**Reply by Zoey2936** [reply]
yes

**Comment by szaimen** [comment, marked as answer]
Will be addressed in https://github.com/nextcloud/all-in-one/pull/3515

**Reply by szaimen** [reply]
This is now released with v7.5.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
